File: javascript/password-validator.js
class PasswordValidator {
    constructor(password) {
        this.password = password;
        this.strength = 0;
    }

    validateLength() {
        if (this.password.length >= 5 && this.password.length <= 9) {
            return true;
        } else if (this.password.length >= 10) {
            this.strength += 2;
            return true;
        } else {
            return false;
        }
    }

    validateSpecialChars(specialChars) {
        if (Array.from(specialChars).some(char => this.password.includes(char))) {
            this.strength += 1;
            return true;
        } else {
            return false;
        }
    }

    validateNumbers(numbersChars) {
        if (Array.from(numbersChars).some(char => this.password.includes(char))) {
            this.strength += 1;
            return true;
        } else {
            return false;
        }
    }

    validateSpaces() {
        if (!this.password.includes(" ")) {
            return true;
        } else {
            return false;
        }
    }

    displayPassword() {
        console.log(`Your entered password: ${this.password}`);
    }

    summary(specialChars, numbersChars) {
        this.displayPassword();
        if (!this.validateSpaces()) {
            console.log("YOUR PASSWORD IS INVALID [CONTAINS SPACES]");
        } else {
            if (this.validateNumbers(numbersChars)) {
                console.log("Password contains numbers.");
            } else {
                console.log("Password does not contain any numbers.");
            }

            if (this.validateSpecialChars(specialChars)) {
                console.log("Password contains at least one special character.");
            } else {
                console.log("Password does not contain any special character.");
            }

            if (this.validateLength()) {
                console.log("Password has a normal or strong length.");
            }

            console.log(`SUMMARY: YOUR PASSWORD STRENGTH IS ${this.strength * 25} / 100%\n`);
        }
    }
}

// Manually input password
const userInputPassword = prompt("Input your password:");

// Define character sets
const LOWERCASE_CHARS = "abcdefghijklmnopqrstuvwxyz";
const UPPERCASE_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
const NUMBERS = "0123456789";
const SYMBOLS = "!@#$%^&*_=+-/";

// Validate the user-input password
const validator = new PasswordValidator(userInputPassword);
validator.summary(SYMBOLS, NUMBERS);

File: java/password-validator.java
import java.util.Scanner;

public class PasswordValidator {
    private String password;
    private int strength;

    public PasswordValidator(String password) {
        this.password = password;
        this.strength = 0;
    }

    public boolean validateLength() {
        int length = password.length();
        if (length >= 5 && length <= 9) {
            strength += 1;
            return true;
        } else if (length >= 10) {
            strength += 2;
            return true;
        } else {
            return false;
        }
    }

    public boolean validateSpecialChars(String specialChars) {
        for (char ch : password.toCharArray()) {
            if (specialChars.contains(String.valueOf(ch))) {
                strength += 1;
                return true;
            }
        }
        return false;
    }

    public boolean validateNumbers(String numbersChars) {
        for (char ch : password.toCharArray()) {
            if (numbersChars.contains(String.valueOf(ch))) {
                strength += 1;
                return true;
            }
        }
        return false;
    }

    public boolean validateSpaces() {
        return !password.contains(" ");
    }

    public void displayPassword() {
        System.out.println("Your entered password: " + password);
    }

    public void summary(String specialChars, String numbersChars) {
        displayPassword();
        if (!validateSpaces()) {
            System.out.println("YOUR PASSWORD IS INVALID [CONTAINS SPACES]");
        } else {
            if (validateNumbers(numbersChars)) {
                System.out.println("Password contains numbers.");
            } else {
                System.out.println("Password does not contain any numbers.");
            }

            if (validateSpecialChars(specialChars)) {
                System.out.println("Password contains at least one special character.");
            } else {
                System.out.println("Password does not contain any special character.");
            }

            if (validateLength()) {
                System.out.println("Password has a normal or strong length.");
            }

            System.out.println("SUMMARY: YOUR PASSWORD STRENGTH IS " + strength * 25 + " / 100%");
        }
    }

    public static void main(String[] args) {
        // Manually input password
        Scanner scanner = new Scanner(System.in);
        System.out.print("Input your password: ");
        String user_input_password = scanner.nextLine();

        // Define character sets
        String LOWERCASE_CHARS = "abcdefghijklmnopqrstuvwxyz";
        String UPPERCASE_CHARS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        String NUMBERS = "0123456789";
        String SYMBOLS = "!@#$%^&*_=+-/";

        // Validate the user-input password
        PasswordValidator validator = new PasswordValidator(user_input_password);
        validator.summary(SYMBOLS, NUMBERS);
    }
}
